src/AvaloniaEdit/Text/TextLineRun.cs

```diff
@@ -350,5 +332,58 @@ private static bool IsSpace(char ch)
         {
             return ch == ' ' || ch == '\u00a0';
         }
+
+        class GlyphWidths
+        {
+            private const double NOT_CALCULATED_YET = -1;
+            private double[] _glyphWidths;
+            private Typeface _typeFace;
+            private double _fontSize;
+            private StringRange _range;
+
+            internal GlyphWidths(StringRange range, Typeface typeFace, double fontSize)
+            {
+                _typeFace = typeFace;
+                _fontSize = fontSize;
+                _range = range;
+
+                _glyphWidths = InitGlyphWidths();
+            }
+
+            internal double GetAt(int index)
+            {
+                if (_glyphWidths[index] == NOT_CALCULATED_YET)
+                    _glyphWidths[index] = MeasureGlyphAt(index);
+
+                return _glyphWidths[index];
+            }
+
+            double MeasureGlyphAt(int index)
+            {
+                return new FormattedText
```

Comment:
I think we should rewrite this to use the current platform's method of measuring a character and avoid constructing a FormattedText instance for every character.

For Skia https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skpaint.getglyphwidths?view=skiasharp-2.80.2#SkiaSharp_SKPaint_GetGlyphWidths_System_ReadOnlySpan_System_Char__

Reply:
Avalonia already supports getting a glyph's width via GlyphTypeface so we might just be able to use that API.

I have added this code lately and it should work for all platforms.

https://github.com/AvaloniaUI/Avalonia/blob/master/src/Skia/Avalonia.Skia/FormattedTextImpl.cs#L715